### PR TITLE
Feat/ot307 18 model migration controlador base users

### PR DESCRIPTION
**Pull request by mrivaslezama**
# Ticket description: feat/OT307-18

Ticket_number# OT307-18

# Checklist:
- [ ] User model creation
- [ ] User migration
- [ ]  Soft deletes added

### CODE REVIEW

**Comment by mrivaslezama** [comment]
Pending AddUserRefToRole user:references, table roles needed first

**Comment by mrivaslezama** [comment]
añadiendo migration RoleIdToUser

File: app/controllers/api/v1/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+      end
+
+      def user_params
+        params.require(:user).permit(:first_name, :last_name, :email, :password, :image)
+      end
+
+      def render_error
+        render json: { errors: @user.errors.full_messages },
+               status: :unprocessable_entity
+      end
```
Comment (fciballero) [inline app/controllers/api/v1/users_controller.rb]: Only the controller with no methods is required here.

File: app/controllers/api/v1/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    end
+  end
+end
```
Comment (fciballero) [inline app/controllers/api/v1/users_controller.rb]: Please don't forget to add an empty line after line 81.

File: app/controllers/application_controller.rb [hunk 3]
```diff
@@ -1,4 +1,10 @@
 # frozen_string_literal: true
 
 class ApplicationController < ActionController::API
+  include Authorizable
+  include Authenticable
+  include Uploadable
+  include Sendeable
+  include Protectable
+  include ActiveStorage::SetCurrent
```
Comment (fciballero) [inline app/controllers/application_controller.rb]: Please check what is the purpose of these lines before adding them.

File: config/routes.rb [hunk 4]
```diff
@@ -1,6 +1,9 @@
 # frozen_string_literal: true
 
 Rails.application.routes.draw do
+
+  resources :sessions, only: [:create]
```
Comment (fciballero) [inline config/routes.rb]: Why did you choose to add this?

**Review by fciballero — COMMENTED** [review]
Looks good, I've left some comments.

**Pull request closed without merging**